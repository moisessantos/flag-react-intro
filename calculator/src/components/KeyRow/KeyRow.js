import React, { memo } from 'react';
import PropTypes from 'prop-types';
import ButtonGroup from 'react-bootstrap/ButtonGroup';

import { KeyButton } from '..';

const KeyRow = ({ values }) => (
  <ButtonGroup size="lg">
    {values.map(value => <KeyButton key={value} text={value} />)}
    <br />
  </ButtonGroup>
);

KeyRow.propTypes = {
  values: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default memo(KeyRow);
