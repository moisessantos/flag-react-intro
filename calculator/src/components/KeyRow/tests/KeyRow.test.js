import React from 'react';
import { shallow } from 'enzyme';
import KeyRow from '..';

describe('KeyRow tests', () => {
    it('should render as expected', () => {
        const wrapper = shallow(<KeyRow values={[1,2,3,4]} />)
        expect(wrapper).toMatchSnapshot();
    });
})