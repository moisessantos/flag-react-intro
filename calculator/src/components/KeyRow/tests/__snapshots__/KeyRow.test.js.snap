// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyRow tests should render as expected 1`] = `
<ButtonGroup
  role="group"
  size="lg"
  toggle={false}
  vertical={false}
>
  <Memo(KeyButton)
    key="1"
    text={1}
  />
  <Memo(KeyButton)
    key="2"
    text={2}
  />
  <Memo(KeyButton)
    key="3"
    text={3}
  />
  <Memo(KeyButton)
    key="4"
    text={4}
  />
  <br />
</ButtonGroup>
`;
